File: backend/src/main/java/com/gpuoj/repository/SubmissionRepository.java
package com.gpuoj.repository;

import com.gpuoj.model.Submission;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

import java.util.UUID;

public interface SubmissionRepository extends ReactiveCrudRepository<Submission, UUID> {
    Flux<Submission> findByUserId(UUID userId);
    Flux<Submission> findByProblemId(String problemId);
}

File: backend/src/main/kotlin/com/gpuoj/repository/SubmissionRepository.kt
package com.gpuoj.repository

import com.gpuoj.model.Submission
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import java.util.UUID

interface SubmissionRepository : ReactiveCrudRepository<Submission, UUID> {
    fun findByUserId(userId: UUID): Flux<Submission>
    fun findByUserIdOrderBySubmittedAtDesc(userId: UUID): Flux<Submission>
    fun findByProblemId(problemId: String): Flux<Submission>
}
